lib/pubsub_client/subscriber.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'google/cloud/pubsub'
+
+module PubsubClient
+  class Subscriber
+    DEFAULT_CONCURRENCY = 8
+
+    # @param subscription [Google::Cloud::PubSub::Subscription]
+    def initialize(subscription)
+      @subscription = subscription
+    end
+
+    # @param concurrency [Integer]
+    # @param auto_ack [Boolean]
+    def subscribe(concurrency, auto_ack, &block)
+      listener = @subscription.listen(threads: { callback: concurrency }) do |received_message|
+        yield received_message.data
+        received_message.acknowledge! if auto_ack
+      end
+
+      begin
+        listener.start
+        sleep
+      rescue SignalException
```

Comment:
In the publisher half of this gem, we use `at_exit` detect when the client is exiting, I believe this is functionally equivalent. We should just have one way of doing this though for consistency's sake.

Reply:
I want to keep it this way and change the publisher to also use a `begin/rescue` block. If you recall, we created a struct to act as our publisher for specs because RSPec was giving us the message
```
The use of doubles or partial doubles from rspec-mocks outside of the per-test lifecycle is not supported. (RSpec::Mocks::OutsideOfExampleError)
```

Keeping it the way I have it here will allow us to use the proper instance doubles instead of creating a struct as we did before. Thoughts?

Reply:
Sounds good!